src/vulkan/engine_vulkan_debugger.cc

```diff
@@ -0,0 +1,976 @@
+// Copyright 2020 The Amber Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/vulkan/engine_vulkan.h"
+
+#if AMBER_ENABLE_VK_DEBUGGING
+
+#include <chrono>              // NOLINT(build/c++11)
+#include <condition_variable>  // NOLINT(build/c++11)
+#include <fstream>
+#include <mutex>  // NOLINT(build/c++11)
+#include <sstream>
+#include <thread>  // NOLINT(build/c++11)
+#include <unordered_map>
+
+#include "dap/network.h"
+#include "dap/protocol.h"
+#include "dap/session.h"
+
+// Set to 1 to enable verbose debugger logging
+#define ENABLE_DEBUGGER_LOG 0
+
+#if ENABLE_DEBUGGER_LOG
+#define DEBUGGER_LOG(...) \
+  do {                    \
+    printf(__VA_ARGS__);  \
+    printf("\n");         \
+  } while (false)
+#else
+#define DEBUGGER_LOG(...)
+#endif
+
+namespace {
+
+// TODO(bclayton): Make script configurable?
```

Comment:
I think this is fine for now. File a bug so we can see if anyone pings it in the future?

Reply:
Okay. Removed for now. You're right - if SwiftShader passes these, I think it's fairly safe to assume timeouts will not be an issue for other drivers. :)